rskj-core/src/test/java/co/rsk/peg/P2shErpFederationTest.java

```diff
@@ -1,68 +1,246 @@
+        networkParameters = bridgeConstants.getBtcParams();
+        emergencyKeys = bridgeConstants.getErpFedPubKeysList();
+        activationDelayValue = bridgeConstants.getErpFedActivationDelay();
+        activations = mock(ActivationConfig.ForBlock.class);
+
+        federation = createDefaultP2shErpFederation();
+    }
 
-        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+    private P2shErpFederation createDefaultP2shErpFederation() {
+        List<FederationMember> standardMembers = FederationTestUtils.getFederationMembersWithBtcKeys(standardKeys);
+        Instant creationTime = ZonedDateTime.parse("2017-06-10T02:30:00Z").toInstant();
+        long creationBlockNumber = 0L;
 
-        Federation p2shErpFederation = new P2shErpFederation(
-            FederationTestUtils.getFederationMembersWithBtcKeys(standardKeys),
-            ZonedDateTime.parse("2017-06-10T02:30:00Z").toInstant(),
-            0L,
-            bridgeConstants.getBtcParams(),
+        return new P2shErpFederation(
+            standardMembers,
+            creationTime,
+            creationBlockNumber,
+            networkParameters,
             emergencyKeys,
-            activationDelay,
+            activationDelayValue,
             activations
         );
+    }
+
+    @Test
+    void createInvalidP2shErpFederation_nullErpKeys() {
+        emergencyKeys = null;
+        assertThrows(FederationCreationException.class, this::createDefaultP2shErpFederation);
+    }
+
+    @Test
+    void createInvalidP2shErpFederation_emptyErpKeys() {
+        emergencyKeys = new ArrayList<>();
+        assertThrows(FederationCreationException.class, this::createDefaultP2shErpFederation);
+    }
+
+    @Test
+    void createValidP2shErpFederation_oneErpKey() {
+        emergencyKeys = Collections.singletonList(emergencyKeys.get(0));
+        assertDoesNotThrow(this::createDefaultP2shErpFederation);
+    }
+
+    @Test
+    void createInvalidP2shErpFederation_negativeCsvValue() {
+        activationDelayValue = -100L;
+        assertThrows(FederationCreationException.class, this::createDefaultP2shErpFederation);
+    }
+
+    @Test
+    void createInvalidP2shErpFederation_zeroCsvValue()  {
+        activationDelayValue = 0L;
+        assertThrows(FederationCreationException.class, this::createDefaultP2shErpFederation);
+    }
+
+    @Test
+    void createInvalidP2shErpFederation_aboveMaxCsvValue()  {
+        activationDelayValue = ErpFederationRedeemScriptParser.MAX_CSV_VALUE + 1;
+        assertThrows(FederationCreationException.class, this::createDefaultP2shErpFederation);
+    }
+
+    @Test
+    void createValidP2shErpFederation_exactMaxCsvValue()  {
+        activationDelayValue = ErpFederationRedeemScriptParser.MAX_CSV_VALUE;
+        assertDoesNotThrow(this::createDefaultP2shErpFederation);
+    }
+
+    @Test
+    void createInvalidFederation_aboveMaxScriptSigSize() {
+        BtcECKey federator9PublicKey = BtcECKey.fromPublicOnly(
```

Comment:
Same as in LegacyErpFederation, let's use the max by default and make it fail by adding only one extra member